src/renderer/screens/account/AccountHeaderActions.js

```diff
@@ -234,6 +241,16 @@ const AccountHeaderActions = ({ account, parentAccount, openModal, t }: Props) =
 
   return (
     <Box horizontal alignItems="center" justifyContent="flex-end" flow={2} mt={15}>
+      {currency.id === "ethereum" ? (
+        <Button small primary onClick={onPlatformStake} event={"Eth Stake Account Button"} icon>
+          <Box horizontal flow={1} alignItems="center">
+            <IconCoins size={14} />
+            <Box>
+              <Trans i18nKey="account.stake" values={{ currency: currency.name }} />
+            </Box>
+          </Box>
+        </Button>
+      ) : null}
```

Comment:
From what I've seen it look likes this whole logic could be implemented in an Ethereum specific `AccountHeaderActions` file and leverage code generation, as done today with algorand, cosmos, polkadot and tron (cf. [here](https://github.com/LedgerHQ/ledger-live-desktop/blob/develop/scripts/sync-families-dispatch.sh#L10) and [here](https://github.com/LedgerHQ/ledger-live-desktop/blob/develop/src/renderer/families/tron/AccountHeaderActions.js) for example)

Reply:
Yes, and no.

this logic doesn't apply to the whole ethereum family but to the ethereum currency only. So it will always need a "if" somewhere.

Never the less, this piece of code will change with the rebranding, and has to be refactored anyway.

Reply:
Also this file already has some "if currency === ethereum" logic, so it's not really making it any worse

Reply:
Your point remain valid, and in the futur we'll have to work on a system where the whole account UI logic is generic but overloadable both by family or by currency.